client/beefy/src/worker.rs

```diff
@@ -912,6 +920,67 @@ where
 			}
 		}
 	}
+
+	/// Report the given equivocation to the BEEFY runtime module. This method
+	/// generates a session membership proof of the offender and then submits an
+	/// extrinsic to report the equivocation. In particular, the session membership
+	/// proof must be generated at the block at which the given set was active which
+	/// isn't necessarily the best block if there are pending authority set changes.
+	pub(crate) fn report_equivocation(
+		&self,
+		proof: EquivocationProof<NumberFor<B>, AuthorityId, Signature>,
+	) -> Result<(), Error> {
+		let rounds =
+			self.persisted_state.voting_oracle.active_rounds().ok_or(Error::UninitSession)?;
+		let (validators, validator_set_id) = (rounds.validators(), rounds.validator_set_id());
+		let offender_id = proof.offender_id().clone();
+
+		if proof.set_id() != validator_set_id {
+			debug!(
+				target: LOG_TARGET,
+				"🥩 Skip equivocation report for old set id: {:?}",
+				proof.set_id()
+			);
+			return Ok(())
```

Comment:
I think I would still report in this case. Why not? We have everything we need to do it, and if the report is already duplicate (very likely in real scenarios), then it will be dropped by `ValidateUnsigned`.